### PR TITLE
Switched to Gradle 7

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

[rn:skip] 

## What does this PR do?

Switch to Gradle 7, this PR has to be applied on top of #13182 and #13183.

- Updates the Gradle wrapper to version `7.2`
- Remove the deprecated `jcenter` and where it was used to retrieve Gradle's plugins it switches to `gradlePluginPortal`
- insert an explicit dependency from test task to the log4j.properties manipulation task used in integration

## Why is it important/What is the impact to the user?

The user is not influenced by this change

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files (and/or docker env variables)~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist

- [ ]

## How to test this PR locally

Simply run a build  task using the wrapped `./gradlew build`

## Related issues

- Relates #13177
- Requires #13182
- Requires #13183

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -601,7 +601,7 @@ project(":logstash-integration-tests") {
         systemProperty 'org.logstash.integration.specs', rubyIntegrationSpecs
         environment "FEATURE_FLAG", System.getenv('FEATURE_FLAG')
         workingDir integrationTestPwd
-        dependsOn installIntegrationTestGems
+        dependsOn = [installIntegrationTestGems, copyProductionLog4jConfiguration]
```
Comment (robbavey) [inline build.gradle:604]: Nit/Question: Is this change necessary given that the `integrationTest` task already defines a `copyProductionLog4jConfiguration` dependency?

**Review by robbavey — APPROVED** [review]
Small question/nit. Feel free to commit as is, if the `copyProductionLog4jConfiguration` dependency is required, or without it removed if it isn't ;)

File: build.gradle (hunk 1 above)
Comment (andsel) [reply build.gradle:604]: Because running `./gradlew clean runIntegrationTests -PrubyIntegrationSpecs=specs/pipeline_log_spec.rb` the dependency on task `copyProductionLog4jConfiguration` seems to get lost and the copying of `log4j2.properties` file to `fixtures/logs_rollover/log4j2.properties` doesn't happen, resulting in:
```
Failures:

      1) Test Logstash Pipeline id rollover of pipeline log file when pipeline.separate_logs is enabled shouldn't create spurious file 
         Failure/Error: log_definition = File.read('fixtures/logs_rollover/log4j2.properties')
         
         Errno::ENOENT:
           No such file or directory - fixtures/logs_rollover/log4j2.properties
         # ./specs/pipeline_log_spec.rb:157:in `block in <main>'
         # /home/andrea/workspace/logstash_andsel/build/qa/integration/vendor/jruby/2.5.0/gems/logstash-devutils-2.2.1-java/lib/logstash/devutils/rspec/spec_helper.rb:51:in `block in <main>'
         # ./rspec.rb:32:in `<main>'

```
